index.test.js

```diff
@@ -608,12 +608,39 @@ test("Save the parent's comment", () => {
   run('a { /*i*/ b {} }', 'a { /*i*/ } a b {}')
 })
 
+test("Save the parent's comment", () => {
+  run(
+    `
+div {
+  /* Comment with ^ $ . | ? * + () */
+  &[data-roots-all^=1] * #id .class {}
+}`,
+    '/* Comment with ^ $ . | ? * + () */ div[data-roots-all^=1] * #id .class {}')
+})
+
+// !
+// test("Save the parent's comment with newline", () => {
+//   run(
+//     `
+// a { 
+//   /*i*/
+
+//   /*i2*/
+//   b {} 
+//   /*i3*/
+//   s {} 
+// }`,
+//     `a { /*i*/ } /*i2*/ a b {} /*i3*/ a s {}`
+//   )
+// })
+
```

Comment:
I added this test to see what happens if there are several rules with comments in the description.

The test does not work - it does not connect selectors for the second nested one. 
I tried to fix it, but in your code, the devil will break my leg. 

Does it make sense to fix it?